parquet-common/src/main/java/org/apache/parquet/bytes/MultiBufferInputStream.java

```diff
@@ -43,15 +42,15 @@ class MultiBufferInputStream extends ByteBufferInputStream {
   private List<ByteBuffer> markBuffers = new ArrayList<>();
 
   MultiBufferInputStream(List<ByteBuffer> buffers) {
-    this.buffers = buffers;
+    List<ByteBuffer> buffersCopy = new ArrayList<>(buffers);
```

Comment:
TBH, this is not needed. It was kind of automatic to copy the list instead of using the one got from outside. But the whole point of these classes is performance. So I'll revert this one.